esrally/mechanic/supplier.py

```diff
@@ -144,7 +144,7 @@ def _supply_requirements(sources, distribution, build, plugins, revisions, distr
                 # * --pipeline=from-sources-skip-build --distribution-version=X.Y.Z where the plugin should not be built but ES should be
                 #   a distributed version.
                 # * --distribution-version=X.Y.Z --revision="my-plugin:abcdef" where the plugin should be built from sources.
-                plugin_needs_build = (sources and build) or distribution
+                plugin_needs_build = build if sources else distribution
```

Comment:
Personally I find the previous styler easier to read and I vote for keeping this and ignoring pylint's suggestion.